### PR TITLE
Feature/mp 2539 add skeleton loading

### PR DESCRIPTION
**Pull request by chopachop**
## What is the purpose of this pull request?
We need a way to indicate that page is being loaded

## What has changed?
Added a placeholder component(skeleton loading) to both VersionsOverview and ProjectsOverview

## Jira links
```jira_links
https://uniwise.atlassian.net/browse/MP-2539
```

### CODE REVIEW

**Review by samirhabibi — APPROVED** [review]

File: dashboard/src/components/Placeholder.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { Box, Skeleton } from "@mui/joy"
+import { FC } from "react"
+
+export const Placeholder: FC = () => {
+  return (
+    <Box sx={{ display: 'flex', flexDirection: 'column', justifyContent: 'space-around' }}>
+      <Skeleton variant="rectangular" width={1750} height={70} sx={{ mt: 10 }} />
+      <Skeleton sx={{ borderRadius: '50%', width: 150, height: 150, mt: 10, ml: 100 }} />
+      <Skeleton variant="rectangular" width={1750} height={70} sx={{ mt: 70 }} />
```
Comment (naueramant) [inline dashboard/src/components/Placeholder.tsx:9]: dynamic widths

File: dashboard/src/views/Projects/index.tsx [hunk 2]
```diff
@@ -53,6 +54,12 @@ export const ProjectsOverview = () => {
     setCurrentPage(newPage);
   };
 
+  if (isProjectsLoading) {
+    return (
+      <Placeholder />
+    )
```
Comment (naueramant) [inline dashboard/src/views/Projects/index.tsx:60]:
```suggestion
    return <Placeholder />;
```

File: dashboard/src/views/Projects/Versions/index.tsx [hunk 3]
```diff
@@ -61,6 +62,12 @@ export const VersionsOverview = () => {
     setCurrentPage(newPage);
   };
 
+  if (isProjectLoading || isVersionsDataLoading) {
+    return (
+      <Placeholder />
+    )
```
Comment (naueramant) [inline dashboard/src/views/Projects/Versions/index.tsx:68]:
```suggestion
    return <Placeholder />;
```

**Review by naueramant — CHANGES_REQUESTED** [review]

**Review by arcuo — APPROVED** [review]

File: dashboard/package.json [hunk 4]
```diff
@@ -27,7 +27,7 @@
     "react-dom": "^18.2.0",
     "react-router": "^6.24.0",
     "react-router-dom": "^6.24.0",
-    "react-toastify": "^10.0.5"
+    "sass": "^1.77.8"
```
Comment (dimitur2204) [inline dashboard/package.json:30]: This should probably be in the devDependencies

File: dashboard/src/api/client/axios.ts [hunk 5]
```diff
@@ -1,21 +1,21 @@
...
+import { ENV } from "../../constants/env";
+import registerLoggerInterceptor from "../interceptors/logger";
 
 // Mock axios instance
 const mockInstance = axios.create();
 registerLoggerInterceptor(mockInstance);
 
 export const mock = new AxiosMockAdapter(mockInstance, {
-  delayResponse: 200,
+  delayResponse: 2000000,
```
Comment (dimitur2204) [inline dashboard/src/api/client/axios.ts:11]: Isn't that a bit much?

File: dashboard/src/views/Projects/components/TableRow/index.tsx [hunk 6]
```diff
@@ -22,7 +21,7 @@ export const ProjectTableRow: FC<ProjectTableRowProps> = ({
   const createdAtDate = formatIsoDateToLocaleString(createdAt);
 
   return (
-    <tr>
+    <tr onClick={() => window.location.href = `/projects/${projectId}/versions`}>
```
Comment (dimitur2204) [inline dashboard/src/views/Projects/components/TableRow/index.tsx:24]: I'm pretty certain that won't be accessible, you have to to make it semantically behave as an `a` tag. Look at [this](https://stackoverflow.com/questions/64776155/make-table-row-or-cell-be-read-by-a-table-as-clickable) stackoverflow post

File: dashboard/src/api/client/axios.ts (hunk 5 above)
Comment (chopachop) [reply dashboard/src/api/client/axios.ts]: yeah, it was rolled back. Was applying some changes yesterday

File: dashboard/src/views/Projects/components/TableRow/index.tsx (hunk 6 above)
Comment (chopachop) [reply dashboard/src/views/Projects/components/TableRow/index.tsx:24]: This is the internal project. In this case, if someone needs to click the row - the button is accessible through the keyboard

**Pull request merged**